src/vis/scatter/utils.ts

```diff
@@ -0,0 +1,257 @@
+    numColorScaleType: ENumericalColorScaleType;
+    shape: ColumnInfo | null;
+    isRectBrush: boolean;
+    alphaSliderVal: number;
+}
+
+const defaultConfig: IScatterConfig = {
+    type: ESupportedPlotlyVis.SCATTER,
+    numColumnsSelected: [],
+    color: null,
+    numColorScaleType: ENumericalColorScaleType.SEQUENTIAL,
+    shape: null,
+    isRectBrush: true,
+    alphaSliderVal: 1
+};
+
+export function scatterMergeDefaultConfig(
+    columns: (NumericalColumn | CategoricalColumn)[],
+    config: IScatterConfig,
+): IVisConfig {
+
+    const merged = merge({}, defaultConfig, config);
+
+    const numCols = columns.filter((c) => c.type === EColumnTypes.NUMERICAL);
+
+    if(merged.numColumnsSelected.length === 0 && numCols.length > 1) {
+        merged.numColumnsSelected.push(numCols[numCols.length - 1].info);
+        merged.numColumnsSelected.push(numCols[numCols.length - 2].info);
+    } else if(merged.numColumnsSelected.length === 1 && numCols.length > 1) {
+        if(numCols[numCols.length - 1].info.id !== merged.numColumnsSelected[0].id) {
+            merged.numColumnsSelected.push(numCols[numCols.length - 1].info);
+        } else {
+            merged.numColumnsSelected.push(numCols[numCols.length - 2].info);
+        }
+    }
+
+    return merged;
+}
+
+const emptyVal = {
+    plots: [],
+    legendPlots: [],
+    rows: 0,
+    cols: 0,
+    errorMessage: 'To create a Scatterplot, please select at least 2 numerical columns.',
+    formList: ['color', 'shape', 'bubble', 'opacity']
+};
+
+export function createScatterTraces(
+    columns: (NumericalColumn | CategoricalColumn)[],
+    selected: {[key: number]: boolean},
+    config: IScatterConfig,
+    scales: Scales,
+    shapes: string[] | null
+): PlotlyInfo {
+    let counter = 1;
+
+    if(!config.numColumnsSelected) {
+        return emptyVal;
+    }
+
+    const validCols: NumericalColumn[] = config.numColumnsSelected.map((c) => columns.filter((col) => col.type === EColumnTypes.NUMERICAL && col.info.id === c.id)[0] as NumericalColumn);
+    const plots: PlotlyData[] = [];
+
+    const shapeScale = config.shape ?
+        d3.scale.ordinal<string>().domain([...new Set((getCol(columns, config.shape) as CategoricalColumn).vals.map((v) => v.val))]).range(shapes)
+        : null;
+
+    let min = 0;
+    let max = 0;
+
+    if(config.color) {
+        min = d3.min((getCol(columns, config.color) as NumericalColumn).vals.map((v) => +v.val).filter((v) => v !== null)),
+        max = d3.max((getCol(columns, config.color) as NumericalColumn).vals.map((v) => +v.val).filter((v) => v !== null));
+    }
+
+    const numericalColorScale = config.color ?
+                d3.scale.linear<string, number>()
+                    .domain([max,
+                        (max + min) / 2,
+                        min])
+                    .range(config.numColorScaleType === ENumericalColorScaleType.SEQUENTIAL ? ['#002245', '#5c84af', '#cff6ff'] : ['#337ab7','#d3d3d3', '#ec6836'])
```

Comment:
Please retrieve the colors from PR https://github.com/phovea/phovea_ui/pull/161/.

Reply:
done except for d3d3d3, since that is not one of our colors